### PR TITLE
Refactor text info target binding

### PR DESCRIPTION
**Pull request by willsb**
Ref #1143 

This adds the more complex target binding, allowing us to use the shortcuts. I ended up reusing a significant amount of iOS code which turned out to be shareable! Nice one. The refactor also allowed me to add tests to the behaviour, making it much more predictable and easier to understand.

For this reason I require functional tests and reviews from members of both teams.

### CODE REVIEW

File: Toggl.Giskard/Autocomplete/TokenSpan.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+using Java.Lang;
+using MvvmCross.Platform;
+using MvvmCross.Platform.Droid;
+using MvvmCross.Plugins.Color.Droid;
+using Toggl.Giskard.Extensions;
+using FoundationColor = Toggl.Foundation.MvvmCross.Helper.Color;
+
+namespace Toggl.Giskard
+{
+    public sealed class ProjectTokenSpan : TokenSpan
```
Comment (kornelijepetak) [inline Toggl.Giskard/Autocomplete/TokenSpan.cs]: Not a big issue, but I would still like every class to live in its own file.

File: Toggl.Giskard/Autocomplete/AutocompleteTextWatcher.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        public event EventHandler TextChanged;
+        public event EventHandler ProjectDeleted;
+        public event EventHandler CursorPositionChanged;
+        public event EventHandler<TagDeletedEventArgs> TagDeleted;
+
+        public void AfterTextChanged(IEditable s)
+        {
+        }
+
+        public void BeforeTextChanged(ICharSequence s, int start, int count, int after)
```
Comment (kornelijepetak) [inline Toggl.Giskard/Autocomplete/AutocompleteTextWatcher.cs]: Would you mind naming the first parameter to something more meaningful, for example `sequence`?

File: Toggl.Giskard/Autocomplete/AutocompleteTextWatcher.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+using Toggl.Foundation.MvvmCross.Autocomplete;
+using Object = Java.Lang.Object;
+
+namespace Toggl.Giskard.Autocomplete
+{
+    public sealed class AutocompleteTextWatcher : Object, ITextWatcher, IAutocompleteEventProvider
+    {
+        public event EventHandler TextChanged;
+        public event EventHandler ProjectDeleted;
+        public event EventHandler CursorPositionChanged;
```
Comment (kornelijepetak) [inline Toggl.Giskard/Autocomplete/AutocompleteTextWatcher.cs]: VS reports this event never used. I guess it's just implemented here to conform to `IAutocompleteEventProvider` interface but the implementor itself does not utilize it. Could you please confirm or check if it should be raised in at least one scenario.

Comment (willsb) [reply Toggl.Giskard/Autocomplete/AutocompleteTextWatcher.cs]: Yes, it's implemented to conform with the interface. So far I haven't had the need to raise this event on Android, no

File: Toggl.Giskard/Autocomplete/AutocompleteTextWatcher.cs (hunk 2 above)
Comment (willsb) [reply Toggl.Giskard/Autocomplete/AutocompleteTextWatcher.cs]: Sure thing!

File: Toggl.Giskard/Autocomplete/TokenSpan.cs (hunk 1 above)
Comment (willsb) [reply Toggl.Giskard/Autocomplete/TokenSpan.cs]: 👌

**Pull request merged**